extensions/2.0/Khronos/KHR_instancing/extensions/2.0/Khronos/KHR_instancing/README.md

```diff
@@ -0,0 +1,64 @@
+# KHR\_instancing
+
+## Khronos 3D Formats Working Group
+
+* Alexey Knyazev
+* Don McCurdy
+* Ed Mackey
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)
+* Romain Guy
+* Tobias Haeussler
+* John Cooke
+
+## Acknowledgments
+
+* TODO
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+TODO  
+
+## Extending Nodes with per instance attributes
+
+Instancing is defined by adding the `KHR_instancing` extension to any glTF mesh node.  The attributes section contains accessor ids for new TRANSFORM4x3 and ID attribute buffers. 
+For example, the following defines some instancing attributes to a mesh node in the graph. Instancing only applies to mesh nodes.  Applying to nodes rather than meshes allows the 
+same mesh to be attached as normal to nodes while also being instanced.  
+
+```json
+{
+    "nodes": [
+        {
+            "mesh": 0,
```

Comment:
One option we should consider is enabling the KHR_instancing extension to (optionally) override the `node.mesh` property. For example:

```js
name: "teapot",
mesh: 0,
extensions: {
  KHR_instancing {
    mesh: 1,
    attributes: { ... }
  }
}
```

This simply provides exporters with some flexibility to control how/if fallback happens for viewers that don't recognize the extension:

1. When the `mesh` override is omitted, clients that don't recognize the extension will render a single instance of the mesh with the node's transform.
2. When the `mesh` override is provided, it should reference a single instance of the mesh. The fallback `node.mesh` value could then point to anything the exporter chooses – a merged mesh of all instances, a single point, or a textured plane saying "404 Not Found".
3. When the `KHR_instancing` extension is marked as required, clients that don't recognize the extension are expected to fail fast without attempting to proceed.